src/main/scala/com/mozilla/telemetry/views/ExperimentAnalysisView.scala

```
@@ -117,35 +135,58 @@ object ExperimentAnalysisView {
   }
 
   def getExperimentMetrics(experiment: String, experimentsSummary: DataFrame, errorAggregates: DataFrame, conf: Conf): List[MetricAnalysis] = {
+    val metadata = ExperimentAnalyzer.getExperimentMetadata(experimentsSummary).collect()
+    val persisted = getPersistedDataFrame(experimentsSummary, metadata, experiment)
+
     val metricList = getMetrics(conf, experimentsSummary)
 
     val metrics = metricList.flatMap {
       case (name: String, md: MetricDefinition) =>
         md match {
           case hd: HistogramDefinition =>
-            new HistogramAnalyzer(name, hd, experimentsSummary).analyze().collect()
+            new HistogramAnalyzer(name, hd, persisted).analyze().collect()
           case sd: ScalarDefinition =>
-            ScalarAnalyzer.getAnalyzer(name, sd, experimentsSummary).analyze().collect()
+            ScalarAnalyzer.getAnalyzer(name, sd, persisted).analyze().collect()
           case _ => throw new UnsupportedOperationException("Unsupported metric definition type")
         }
     }
 
-    val metadata = ExperimentAnalyzer.getExperimentMetadata(experimentsSummary).collect()
     val crashes = CrashAnalyzer.getExperimentCrashes(errorAggregates)
     (metadata ++ metrics ++ crashes).toList
   }
 
+  // Adds a column for permutations, repartitions if warranted, and persists the result for quicker access
+  def getPersistedDataFrame(experimentsSummary: DataFrame, metadata: Array[MetricAnalysis], experiment: String): DataFrame = {
+    val topLevelMetadata = metadata.filter(_.subgroup == "All")
+    val totalPings = topLevelMetadata.flatMap(_.statistics.get.filter(_.name == "Total Pings").map(_.value)).sum
+    val branchCounts = topLevelMetadata.map {
+      r => r.experiment_branch ->
+        (r.statistics.get.collectFirst {case s: Statistic if s.name == "Total Clients" => s} match {
+          case Some(s: Statistic) => s.value.toLong
+          case _ => 0L
```

Is this case expected? Should we raise an error instead?